api/views.py

```
@@ -1,136 +0,0 @@
-from django.http import Http404
-from rest_framework import generics, status
-from rest_framework.permissions import IsAuthenticated
-from rest_framework.response import Response
-from rest_framework.views import APIView
-from api.models import PreviousProject, User
-from api.serializers import PreviousProjectSerializer, UserSerializer
-from rest_framework.permissions import AllowAny
-from api.permissions import IsLoggedInUserOrAdmin, IsAdminUser, ReadOnly
-from django.utils import timezone
-
-
-class HomeListView(generics.ListAPIView):
-    http_method_names = ['get']
-    serializer_class = PreviousProjectSerializer
-    queryset = PreviousProject.objects.filter(
-        pub_date__lte=timezone.now()
-    )
-    permission_classes = [IsAuthenticated | ReadOnly]
-
-
-# get all previous projects
-
-
-class HomeAPIView(generics.CreateAPIView):
-    http_method_names = ['post']
-    serializer_class = PreviousProjectSerializer
-    permission_classes = [IsAuthenticated]
-
-    def post(self, request, *args, **kwargs):
-        serializer = PreviousProjectSerializer(data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            content = {
-                'data': serializer.data,
-                'msg': 'project created successfully',
-                'status': status.HTTP_201_CREATED
-            }
-            return Response(content)
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

instead if serializer  is not Valid you can raise a validation error with serializer errors so the client can know which field is missing or what is  wrong with the request the client sent